File: DSA/Ayush_Mishra_590014844/Day_12_Question_3.py
def compress_string(s):
    compressed = []
    count = 1
    for i in range(1, len(s) + 1):
        if i < len(s) and s[i] == s[i - 1]:
            count += 1
        else:
            compressed.append(f"{s[i - 1]}{count}")
            count = 1
    compressed_str = ''.join(compressed)
    return compressed_str if len(compressed_str) < len(s) else s

# Example usage
print(compress_string("aabcccccaaa"))  # a2b1c5a3

File: DSA/Ayush_Mishra_590014844/Day_12_Question_3.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
void compressString(const char *str) {
    int len = strlen(str), count = 1;
    char buffer[100] = "";
    int j = 0;

    for (int i = 1; i <= len; i++) {
        if (str[i] == str[i - 1]) {
            count++;
        } else {
            j += sprintf(buffer + j, "%c%d", str[i - 1], count);
            count = 1;
        }
    }

    if (strlen(buffer) >= len)
        printf("%s\n", str);
    else
        printf("%s\n", buffer);
}

int main() {
    compressString("aabcccccaaa");  // a2b1c5a3
    return 0;
}
